examples/thermostat/thermostat-common/src/thermostat-delegate-impl.cpp

```diff
@@ -347,52 +339,61 @@ CHIP_ERROR ThermostatDelegate::RemoveFromThermostatSuggestionsList(uint8_t uniqu
     return CHIP_NO_ERROR;
 }
 
-uint8_t ThermostatDelegate::GetUniqueID()
+bool ThermostatDelegate::FindInList(uint8_t uniqueIDToFind)
+{
+    for (size_t index = 0; index < static_cast<size_t>(mNextFreeIndexInThermostatSuggestionsList); index++)
+    {
+        if (uniqueIDToFind == mThermostatSuggestions[index].GetUniqueID())
+        {
+            return true;
+        }
+    }
+    return false;
```

Comment:
Perhaps consider:
```suggestion
    for (auto & suggestion : Span(mThermostatSuggestions, mNextFreeIndexInThermostatSuggestionsList))
    {
        if (uniqueIDToFind == suggestion.GetUniqueID())
        {
            return true;
        }
    }
    return false;
```

but I don't feel too strongly about this.

If desired, we could also have a method returning `Span(mThermostatSuggestions, mNextFreeIndexInThermostatSuggestionsList)`, something like `CurrentSuggestions()` or something?